proposals/lip-0061.md

```diff
@@ -520,10 +523,10 @@ def selectAggregateCommit() -> AggregateCommit:
     # Note that the BFT parameters at height maxHeightCertified+1 are already authenticated by the previous certificate.
     heightNextBFTParameters = getNextHeightBFTParameters(getBFTHeights().maxHeightCertified + 1)
     if previous function call returns valid height:
-        nextHeight = min(heightNextBFTParameters-1, getBFTHeights().maxHeightPrecommitted)
+        nextHeight = min(max(heightNextBFTParameters -1, MIN_CERTIFICATE_HEIGHT), getBFTHeights().maxHeightPrecommitted)
```

Comment:
`MIN_CERTIFICATE_HEIGHT` seems to be the height of the first block of the round, but `heightNextBFTParameters -1` is the last block of the round. isn't it bit inconsistent?

Reply:
I think what we want here is that in the while loop below, `nextHeight` always has a value of at least `MIN_CERTIFICATE_HEIGHT`, i.e., this is the minimum height for which we generate a certificate. I don't think the current logic guarantees that.

Reply:
@shuse2 It is not matter of consistency, but feasibility. The `MIN_CERTIFICATE_HEIGHT` has to be first block of a round, since this is the first block where bft params are guaranteed to exist. For `heightNextBFTParameters -1` the point is that we want to certify the block that decides the new validators and this is the last block of round by construction of PoS, so I think this can not be changed.

Reply:
@janhack in case `heightNextBFTParameters` is undefined, the value of `nextHeight` will not be greater than `MIN_CERTIFICATE_HEIGHT`. But somehow we don't care about this case; even if an aggregate commit could be created, it would be rejected.

On the other hand, we really need the max() in case `heightNextBFTParameters` defined, in order to be able to create valid aggregate commits.

Reply:
I updated the usage of max operators (reverting to a previous version) since it seemed confusing for everyone.